versioned/mongodb/src/test/java/com/dremio/nessie/versioned/store/mongodb/codecs/TestBsonToEntityConverter.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dremio.nessie.versioned.store.mongodb.codecs;
+
+import java.util.Map;
+
+import org.bson.BsonReader;
+import org.bson.json.JsonReader;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import com.dremio.nessie.versioned.store.Entity;
+import com.google.common.collect.ImmutableMap;
+
+public class TestBsonToEntityConverter {
+  @Test
+  public void readNoDocumentRoot() {
+    final BsonReader reader = new JsonReader("1");
+    Assertions.assertThrows(UnsupportedOperationException.class, () -> BsonToEntityConverter.read(reader));
```

Comment:
Let's come up with a more specific error message for decoding/unexpected object failures.